lib/ansible/executor/task_executor.py

```diff
@@ -252,7 +252,11 @@ def _execute(self, variables=None):
         # variable not being present which would otherwise cause validation to fail
         if not self._task.evaluate_conditional(templar, variables):
             debug("when evaulation failed, skipping this task")
-            return dict(changed=False, skipped=True, skip_reason='Conditional check failed')
+            skip_result = dict(changed=False, skipped=True, skip_reason='Conditional check failed')
+            # propagate no_log into the result if necessary
+            if self._play_context.no_log:
+                skip_result['_ansible_no_log'] = True
+            return skip_result
```

Comment:
this one is not really needed as there is no user data passed back in skip_result